Applications/WebPortal/WebApp/library/source/components/number_input.tsx

```diff
@@ -0,0 +1,109 @@
+import { css, StyleSheet } from 'aphrodite';
+import * as React from 'react';
+
+interface Properties {
+
+  /** The value to display in the field. */
+  value?: number;
+
+  /** Additional CSS styles. */
+  style?: any;
+
+  /** The class name of the input box. */
+  className?: string;
+
+  /** The smallest number that the box accepts. */
+  min?: number;
+
+  /** The largest number that the box accepts. */
+  max?: number;
+
+  /** Called when the value changes.
+   * @param value - The updated value.
+   */
+  onChange?: (value: number) => void;
+}
+
+/** Displays a single number input field. */
+export class NumberInput extends React.Component<Properties> {
+  public static readonly defaultProps = {
+    value: 0,
+    onChange: () => {}
+  };
+
+  public render(): JSX.Element {
+    return (
+      <input 
+        type={'number'}
+        min={this.props.min}
+        max={this.props.max}
+        value={this.props.value}
+        style={{...NumberInput.STYLE.box, ...this.props.style}}
+        onChange={this.onChange.bind(this)}
```

Comment:
This will cause a ```bind``` to happen on every call to ```render```. You will need to instead do the ```bind``` in the constructor.

Reply:
I think in this case... if I inline it I don't need to bind it at all.